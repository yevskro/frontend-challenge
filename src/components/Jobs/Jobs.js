import React from 'react';
import styled from 'styled-components';

function Jobs(){
  return <StyledJobs role='main'>

  </StyledJobs>
}

const StyledJobs = styled.section`
  position: absolute;
  z-index: 2;
  border: 1px solid red;
  left: 5%;
  right: 5%;
  top: 120px;
`
export default Jobs;