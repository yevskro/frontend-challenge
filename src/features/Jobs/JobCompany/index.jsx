import React from 'react';
import styled from 'styled-components';

function JobCompany({ children, newly, featured }) {
  return (
    <StyledJobCompany role="heading">
      {children}
      {newly ? <PrimaryPill role="note">NEW!</PrimaryPill> : ''}
      {featured ? <SecondaryPill role="note">FEATURED</SecondaryPill> : ''}
    </StyledJobCompany>
  );
}

const Pill = styled.span`
  line-height: 26px;
  vertical-align: middle;
  height: 24px;
  padding-left: 9px;
  padding-right: 9px;
  color: white;
  border-radius: 20px;
  margin-left: 4px;
  margin-right: 4px;
  font-size: ${({ theme }) => theme.fontSize.small};
`;

const PrimaryPill = styled(Pill)`
  background-color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
  margin-left: 18px;
  font-size: ${({ theme }) => theme.fontSize.xxsmall};
`;

const SecondaryPill = styled(Pill)`
  background-color: ${({ theme }) => theme.color.neutral.veryDarkGrayishCyan};
  font-size: ${({ theme }) => theme.fontSize.xxsmall};
`;

const StyledJobCompany = styled.div`
  display: flex;
  color: #5eaaa6;
  font-weight: ${({ theme }) => theme.fontWeight.bold};
  height: 24px;
  align-items: center;
  font-size: ${({ theme }) => theme.fontSize.xsmall};
  @media (min-width: 500px) {
    font-size: ${({ theme }) => theme.fontSize.small};
  }
`;
export default JobCompany;
