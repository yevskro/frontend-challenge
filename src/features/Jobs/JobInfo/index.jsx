import React from 'react';
import styled from 'styled-components';

function JobInfo({ postedAt, contract, location }) {
  return (
    <StyledJobInfo>
      <Info>{postedAt}</Info>
      <Info>{contract}</Info>
      <Info>{location}</Info>
    </StyledJobInfo>
  );
}

const Info = styled.li`
  white-space: nowrap; /* no wrap arounds to the next line */

  /* increasing the size of the bullet represented by content character '.' */
  &:first-child::before {
    /* 
      no bullet point before the first element fixing the font-size so the element
      does not jump
    */
    font-size: 1.7rem;
    content: ' ';
  }

  &:not(:first-child)::before {
    display: inline-block;
    content: '.';
    padding: 0px 4px 0px 4px;
    font-size: 1.7rem;
    transform: translateY(-16%); /* moves the . to the vertical center */

    @media (min-width: 600px) {
      /* increases the space between the point and the text around */
      padding: 0px 8px 0px 8px;
    }
  }

  @media (min-width: 600px) {
    /* works for tablet and desktop views */
    font-size: ${({ theme }) => theme.fontSize.medium};
  }
`;

const StyledJobInfo = styled.ul`
  display: flex;
  color: ${({ theme }) => theme.color.neutral.darkGrayishCyan};
  font-weight: ${({ theme }) => theme.fontWeight.thin};
  font-size: ${({ theme }) => theme.fontSize.small};
`;

export default JobInfo;
