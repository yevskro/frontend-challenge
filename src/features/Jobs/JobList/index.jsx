import React from 'react';
import styled from 'styled-components';

function JobList({ children }) {
  return <StyledJobList>{children}</StyledJobList>;
}

const StyledJobList = styled.ul`
  margin-bottom: 150px;
`;

export default JobList;
