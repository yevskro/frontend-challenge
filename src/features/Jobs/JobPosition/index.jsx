import React from 'react';
import styled from 'styled-components';
import JobInfo from '../JobInfo';
import JobCompany from '../JobCompany';

function JobPosition({ children, postedAt, contract, location, company }) {
  return (
    <StyledJobPosition>
      <JobCompany newly={company.newly} featured={company.featured}>
        {company.name}
      </JobCompany>
      <Heading>{children}</Heading>
      <JobInfo postedAt={postedAt} contract={contract} location={location} />
    </StyledJobPosition>
  );
}

const Heading = styled.h3`
  padding: 11px 0px 11px 0px;
  margin: 0px;
  white-space: nowrap;
  font-weight: ${({ theme }) => theme.fontWeight.bold};

  &:hover {
    color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
    cursor: pointer;
  }
`;

const StyledJobPosition = styled.div`
  display: flex;
  flex-direction: column;

  @media (min-width: 500px) {
    font-size: ${({ theme }) => theme.fontSize.medium};
  }

  @media (min-width: 600px) {
    font-size: ${({ theme }) => theme.fontSize.large};
  }
`;

export default JobPosition;
