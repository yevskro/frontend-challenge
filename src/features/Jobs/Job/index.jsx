import React from 'react';
import styled from 'styled-components';
import JobPosition from '../JobPosition';
import SearchFilters from '../../SearchByFilters/SearchFilters';
import SearchFilter from '../../SearchByFilters/SearchFilter';

function Job({ data, onAddFilter }) {
  function jsxFilters() {
    /* creates SearchFilters based on data information */
    const filters = [];

    filters.push(
      <SearchFilter
        onAddFilter={onAddFilter}
        filter={{ role: data.role }}
        key={`job-item-filter-${data.role}`}
      >
        {data.role}
      </SearchFilter>
    );

    filters.push(
      <SearchFilter
        onAddFilter={onAddFilter}
        filter={{ level: data.level }}
        key={`job-item-filter-${data.level}`}
      >
        {data.level}
      </SearchFilter>
    );

    data.languages.forEach((language) => {
      filters.push(
        <SearchFilter
          onAddFilter={onAddFilter}
          filter={{ languages: language }}
          key={`job-item-filter-${language}`}
        >
          {language}
        </SearchFilter>
      );
    });

    data.tools.forEach((tool) => {
      filters.push(
        <SearchFilter
          onAddFilter={onAddFilter}
          filter={{ tools: tool }}
          key={`job-item-filter-${tool}`}
        >
          {tool}
        </SearchFilter>
      );
    });

    return filters;
  }

  return (
    <StyledJob featured={data.featured} role="listitem">
      <JobInformation>
        <CompanyLogo src={data.logo} alt="company logo" />
        <JobPosition
          company={{
            name: data.company,
            newly: data.newly,
            featured: data.featured,
          }}
          postedAt={data.postedAt}
          contract={data.contract}
          location={data.location}
        >
          {data.position}
        </JobPosition>
      </JobInformation>
      <HorizontalLine />
      <JobFilters>
        <SearchFilters flexEnd>{jsxFilters()}</SearchFilters>
      </JobFilters>
    </StyledJob>
  );
}

const JobFilters = styled.div`
  display: flex;
  align-items: center;
  margin-left: auto;
`;

const HorizontalLine = styled.hr`
  opacity: 0.5;
  margin: 0px 0px 4px 0px;
  border-top: 1px solid ${({ theme }) => theme.color.neutral.darkGrayishCyan};
  @media (min-width: 785px) {
    /* desktop view */
    display: none;
  }
`;

const JobInformation = styled.div`
  display: flex;
  align-items: center;
  padding-bottom: 10px;
  @media (min-width: 785px) {
    /* desktop view */
    padding-bottom: 0px;
  }
`;

const CompanyLogo = styled.img`
  position: absolute;
  height: 48px;
  width: 48px;
  transform: translate(
    0px,
    -86px
  ); /* translate the icon to align half way above the container top border */
  @media (min-width: 555px) {
    /* tablet view */
    position: relative;
    height: 88px;
    width: 88px;
    margin-right: 24px;
    transform: translate(0, 0);
  }
`;

const StyledJob = styled.div`
  box-sizing: border-box;
  width: 100%;
  margin-top: 40px;
  padding: 44px 24px 14px 20px;
  background-color: white;
  border-radius: 5px;
  border-left: 5px solid
    ${(props) =>
      props.featured ? props.theme.color.primary.desaturtedDarkCyan : 'white'};
  box-shadow: 0px 10px 10px ${({ theme }) => theme.color.shadow};

  @media (min-width: 555px) {
    /* tablet view */
    display: flex;
    flex-direction: column;
    padding-top: 20px;
  }

  @media (min-width: 785px) {
    /* desktop view */
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 24px 24px 24px 24px;
  }

  @media (min-width: 900px) {
    /* fits it nicer on a bigger desktop view */
    padding: 30px 48px 30px 48px;
  }
`;

export default Job;
