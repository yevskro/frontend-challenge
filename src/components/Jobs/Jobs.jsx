import React from 'react';
import styled from 'styled-components';
import Filters from '../Filters/Filters';
import Filter from '../Filter/Filter';
import Search from '../Search/Search';
import ClearButton from '../ClearButton/ClearButton';
import JobList from '../JobList/JobList';
import Job from '../Job/Job';
import jobData from '../../data.json';

function Jobs() {
  return (
    <StyledJobs role='main'>
      <Search>
        <Filters>
          <Filter remove>Frontend</Filter>
          <Filter remove>CSS</Filter>
          <Filter remove>JavaScript</Filter>
        </Filters>
        <ClearButton/>
      </Search>
      <JobList>
        <Job data={jobData[0]} />
        <Job data={jobData[1]} />
        <Job data={jobData[2]} />
      </JobList>
    </StyledJobs>
  );
}

const StyledJobs = styled.section`
  position: absolute;
  z-index: 2;
  left: 5%;
  right: 5%;
  top: 120px;
  background: transparent;
  @media (min-width: 401px) {
    left: 11%;
    right: 11%;
  }
`;

export default Jobs;
