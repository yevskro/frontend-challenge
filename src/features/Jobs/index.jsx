import React from 'react';
import styled from 'styled-components';
import SearchByFilters from '../SearchByFilters';
import Job from './Job';
import JobList from './JobList';
import jobData from '../../data.json';
import SearchFilters from '../SearchByFilters/SearchFilters';
import SearchFilter from '../SearchByFilters/SearchFilter';
import useSearchByFilters from '../SearchByFilters/useSearchByFilters';

function Jobs() {
  const { filteredData, filters, addFilter, removeFilter } = useSearchByFilters(
    jobData
  );

  const jsxFilters = filters.map((filter) => (
    <SearchFilter filter={filter} remove onRemoveFilter={removeFilter}>
      {filter[Object.keys(filter)[0]]}
    </SearchFilter>
  ));

  const jsxJobs = filteredData.map((job) => (
    <Job data={job} onAddFilter={addFilter} />
  ));

  return (
    <StyledJobs role="main">
      <SearchByFilters visible={!!filters.length}>
        <SearchFilters>{jsxFilters}</SearchFilters>
      </SearchByFilters>
      <JobList>{jsxJobs}</JobList>
    </StyledJobs>
  );
}

const StyledJobs = styled.section`
  position: absolute;
  z-index: 2;
  left: 5%;
  right: 5%;
  top: 120px;
  background: transparent;
  @media (min-width: 401px) {
    left: 11%;
    right: 11%;
  }
`;

export default Jobs;
