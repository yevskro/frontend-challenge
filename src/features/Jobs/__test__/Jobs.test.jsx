import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Theme from '../../../shared/providers/Theme';
import jobData from '../../../shared/test/test-data.json';
import Jobs from '..';

test('renders', () => {
  render(
    <Theme>
      <Jobs jobs={[jobData[0], jobData[1]]} />
    </Theme>
  );

  const searchElement = screen.getByRole('search');
  expect(screen.getByRole('search')).toBeInTheDocument();
  expect(searchElement).toHaveStyleRule('visibility', 'hidden');
  const jobElements = screen.getAllByRole('img', { name: 'company logo' });
  expect(jobElements.length).toBe(2);
  expect(jobElements[0]).toBeInTheDocument();
  expect(jobElements[1]).toBeInTheDocument();
});

test('search component appears when applying the first filter', () => {
  render(
    <Theme>
      <Jobs jobs={[jobData[0], jobData[1]]} />
    </Theme>
  );

  const htmlFilterElement = screen.getByText('HTML');
  expect(htmlFilterElement).toBeInTheDocument();
  userEvent.click(htmlFilterElement);
  expect(screen.getByRole('search')).toHaveStyleRule('visibility', 'visible');
});

test('jobs get filtered by filter', () => {
  render(
    <Theme>
      <Jobs jobs={[jobData[0], jobData[1]]} />
    </Theme>
  );

  const htmlFilterElement = screen.getByText('HTML');
  expect(htmlFilterElement).toBeInTheDocument();
  userEvent.click(htmlFilterElement);
  const jobElements = screen.getAllByRole('img', { name: 'company logo' });
  expect(jobElements.length).toBe(1);
  expect(jobElements[0]).toHaveAttribute('src', '/assets/images/photosnap.svg');
  expect(jobElements[0]).toBeInTheDocument();
});

test('jobs get unfiltered after removing a filter', () => {
  render(
    <Theme>
      <Jobs jobs={[jobData[0], jobData[1]]} />
    </Theme>
  );

  const htmlFilterElement = screen.getByText('HTML');
  expect(htmlFilterElement).toBeInTheDocument();
  userEvent.click(htmlFilterElement);
  let jobElements = screen.getAllByRole('img', { name: 'company logo' });
  expect(jobElements.length).toBe(1);
  expect(jobElements[0]).toBeInTheDocument();
  const unfilterElement = screen.getByRole('img', { name: 'remove icon' });
  expect(unfilterElement).toBeInTheDocument();
  userEvent.click(unfilterElement);
  jobElements = screen.getAllByRole('img', { name: 'company logo' });
  expect(jobElements.length).toBe(2);
  expect(jobElements[0]).toHaveAttribute('src', '/assets/images/photosnap.svg');
  expect(jobElements[0]).toBeInTheDocument();
  expect(jobElements[1]).toHaveAttribute('src', '/assets/images/manage.svg');
  expect(jobElements[1]).toBeInTheDocument();
});

test('jobs original state is restored after clearing filters', () => {
  render(
    <Theme>
      <Jobs jobs={[jobData[0], jobData[1]]} />
    </Theme>
  );

  const htmlFilterElement = screen.getByText('HTML');
  expect(htmlFilterElement).toBeInTheDocument();
  userEvent.click(htmlFilterElement);
  let jobElements = screen.getAllByRole('img', { name: 'company logo' });
  expect(jobElements.length).toBe(1);
  expect(jobElements[0]).toBeInTheDocument();
  const clearElement = screen.getByRole('button', { name: 'Clear' });
  expect(clearElement).toBeInTheDocument();
  userEvent.click(clearElement);
  jobElements = screen.getAllByRole('img', { name: 'company logo' });
  expect(jobElements.length).toBe(2);
  expect(jobElements[0]).toHaveAttribute('src', '/assets/images/photosnap.svg');
  expect(jobElements[0]).toBeInTheDocument();
  expect(jobElements[1]).toHaveAttribute('src', '/assets/images/manage.svg');
  expect(jobElements[1]).toBeInTheDocument();
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <Theme>
        <Jobs jobs={[jobData[0], jobData[1]]} />
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
