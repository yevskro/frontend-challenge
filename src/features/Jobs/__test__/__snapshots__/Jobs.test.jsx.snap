// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c3 {
  text-align: center;
  font-size: 1rem;
  color: hsl(180,8%,52%);
}

.c3:hover {
  color: hsl(180,29%,50%);
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 65px;
  padding: 5px 26px 5px 21px;
  margin-bottom: 56px;
  border-radius: 5px;
  background-color: #fff;
  box-shadow: 0px 10px 10px rgba(94,140,160,0.15);
  visibility: hidden;
}

.c15 {
  white-space: nowrap;
}

.c15:first-child::before {
  font-size: 1.7rem;
  content: ' ';
}

.c15:not(:first-child)::before {
  display: inline-block;
  content: '.';
  padding: 0px 4px 0px 4px;
  font-size: 1.7rem;
  -webkit-transform: translateY(-16%);
  -ms-transform: translateY(-16%);
  transform: translateY(-16%);
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: hsl(180,8%,52%);
  font-weight: 400;
  font-size: 0.9rem;
}

.c10 {
  height: 24px;
  line-height: 26px;
  vertical-align: middle;
  padding: 0px 9px 0px 9px;
  margin: 0px 4px 0px 4px;
  color: white;
  border-radius: 20px;
  font-size: 0.9rem;
}

.c11 {
  margin-left: 18px;
  background-color: hsl(180,29%,50%);
  font-size: 0.7rem;
}

.c12 {
  background-color: hsl(180,14%,20%);
  font-size: 0.7rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 24px;
  color: hsl(180,29%,50%);
  font-weight: 600;
  font-size: 0.8rem;
}

.c13 {
  padding: 11px 0px 11px 0px;
  margin: 0px;
  white-space: nowrap;
  font-weight: 600;
}

.c13:hover {
  color: hsl(180,29%,50%);
  cursor: pointer;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  margin: 10px 16px 10px 0px;
  border-radius: 5px;
  overflow: hidden;
  background-color: hsl(180,31%,95%);
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  padding: 0px 9px 0px 8px;
  color: hsl(180,29%,50%);
  font-size: 0.9rem;
  font-weight: 600;
}

.c20:hover {
  color: white;
  background-color: hsl(180,29%,50%);
  cursor: pointer;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-left: auto;
}

.c16 {
  opacity: 0.5;
  margin: 0px 0px 4px 0px;
  border-top: 1px solid hsl(180,8%,52%);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 10px;
}

.c7 {
  position: absolute;
  height: 48px;
  width: 48px;
  -webkit-transform: translate(0px,-86px);
  -ms-transform: translate(0px,-86px);
  transform: translate(0px,-86px);
}

.c5 {
  box-sizing: border-box;
  width: 100%;
  margin-top: 40px;
  padding: 44px 24px 14px 20px;
  background-color: white;
  border-radius: 5px;
  border-left: 5px solid hsl(180,29%,50%);
  box-shadow: 0px 10px 10px rgba(94,140,160,0.15);
}

.c4 {
  margin-bottom: 150px;
}

.c0 {
  position: absolute;
  z-index: 2;
  left: 5%;
  right: 5%;
  top: 120px;
  background: transparent;
}

@media (min-width:600px) {
  .c15:not(:first-child)::before {
    padding: 0px 8px 0px 8px;
  }
}

@media (min-width:600px) {
  .c15 {
    font-size: 1rem;
  }
}

@media (min-width:500px) {
  .c9 {
    font-size: 0.9rem;
  }
}

@media (min-width:500px) {
  .c8 {
    font-size: 1rem;
  }
}

@media (min-width:600px) {
  .c8 {
    font-size: 1.2rem;
  }
}

@media (max-width:415px) {
  .c2 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:415px) {
  .c18 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:415px) {
  .c19 {
    margin: 10px 0px 10px 16px;
  }
}

@media (min-width:785px) {
  .c16 {
    display: none;
  }
}

@media (min-width:785px) {
  .c6 {
    padding-bottom: 0px;
  }
}

@media (min-width:555px) {
  .c7 {
    position: relative;
    height: 88px;
    width: 88px;
    margin-right: 24px;
    -webkit-transform: translate(0,0);
    -ms-transform: translate(0,0);
    transform: translate(0,0);
  }
}

@media (min-width:555px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    padding-top: 20px;
  }
}

@media (min-width:785px) {
  .c5 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    padding: 24px 24px 24px 24px;
  }
}

@media (min-width:900px) {
  .c5 {
    padding: 30px 48px 30px 48px;
  }
}

@media (min-width:401px) {
  .c0 {
    left: 11%;
    right: 11%;
  }
}

<section
  className="c0"
  role="main"
>
  <div
    className="c1"
    role="search"
  >
    <ul
      className="c2"
    />
    <button
      className="c3"
      onClick={[Function]}
    >
      Clear
    </button>
  </div>
  <ul
    className="c4"
  >
    <div
      className="c5"
      role="listitem"
    >
      <div
        className="c6"
      >
        <img
          alt="company logo"
          className="c7"
          src="/assets/images/photosnap.svg"
        />
        <div
          className="c8"
        >
          <div
            className="c9"
          >
            <h6>
              Photosnap
            </h6>
            <span
              className="c10 c11"
            >
              NEW!
            </span>
            <span
              className="c10 c12"
            >
              FEATURED
            </span>
          </div>
          <h3
            className="c13"
          >
            Senior Frontend Developer
          </h3>
          <ul
            className="c14"
          >
            <li
              className="c15"
            >
              2d ago
            </li>
            <li
              className="c15"
            >
              Part Time
            </li>
            <li
              className="c15"
            >
              USA Only
            </li>
          </ul>
        </div>
      </div>
      <hr
        className="c16"
      />
      <div
        className="c17"
      >
        <ul
          className="c18"
        >
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              Frontend
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              Senior
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              HTML
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              CSS
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              JavaScript
            </span>
            
          </li>
        </ul>
      </div>
    </div>
    <div
      className="c5"
      role="listitem"
    >
      <div
        className="c6"
      >
        <img
          alt="company logo"
          className="c7"
          src="/assets/images/manage.svg"
        />
        <div
          className="c8"
        >
          <div
            className="c9"
          >
            <h6>
              Manage
            </h6>
            <span
              className="c10 c11"
            >
              NEW!
            </span>
            <span
              className="c10 c12"
            >
              FEATURED
            </span>
          </div>
          <h3
            className="c13"
          >
            Fullstack Developer
          </h3>
          <ul
            className="c14"
          >
            <li
              className="c15"
            >
              1d ago
            </li>
            <li
              className="c15"
            >
              Part Time
            </li>
            <li
              className="c15"
            >
              Remote
            </li>
          </ul>
        </div>
      </div>
      <hr
        className="c16"
      />
      <div
        className="c17"
      >
        <ul
          className="c18"
        >
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              Fullstack
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              Midweight
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              Python
            </span>
            
          </li>
          <li
            className="c19"
          >
            <span
              className="c20"
              onClick={[Function]}
            >
              React
            </span>
            
          </li>
        </ul>
      </div>
    </div>
  </ul>
</section>
`;
